book/src/dev/rfcs/drafts/xxxx-async-rust.md

```diff
@@ -0,0 +1,438 @@
+
+## Awaiting Multiple Futures
+[awaiting-multiple-futures]: #awaiting-multiple-futures
+
+When awaiting multiple futures, Zebra can use biased or unbiased selection.
+
+Typically, we prefer unbiased selection, so that if multiple futures are ready,
+they each have a chance of completing. But if one of the futures needs to take
+priority (for example, cancellation), you might want to use biased selection.
+
+### Unbiased Selection
+[unbiased-selection]: #unbiased-selection
+
+The [`futures::select!`](https://docs.rs/futures/0.3.13/futures/macro.select.html) and
+[`tokio::select!`](https://docs.rs/tokio/0.3.6/tokio/macro.select.html) macros select
+ready arguments at random.
+
+Also consdier the `FuturesUnordered` stream for unbiased selection of a large number
+of futures. However, this macro and stream require mapping all arguments to the same
+type.
+
+Consider mapping the returned type to a custom enum with module-specific names.
+
+### Biased Selection
+[biased-selection]: #biased-selection
+
+The [`futures::select`](https://docs.rs/futures/0.3.13/futures/future/fn.select.html)
+is biased towards its first argument. If the first argument is always ready, the second
+argument will never be returned. (This behavior is not documented or guaranteed.) This
+bias can cause starvation or hangs. Consider edge cases where queues are full, or there
+are a lot of messages. If in doubt:
+- put shutdown or cancel oneshots first, then timers, then other futures
+- use the `select!` macro to ensure fairness
+
+Select's bias can be useful to ensure that cancel oneshots and timers are always
+executed first. Consider the `select_biased!` macro and `FuturesOrdered` stream
+for guaranteed ordered selection of futures. (However, this macro and stream require
+mapping all arguments to the same type.)
+
+The `futures::select` `Either` return type is complex, particularly when nested. This
+makes code hard to read and maintain. Map the `Either` to a custom enum.
+
+## Testing Async Code
+[testing-async-code]: #testing-async-code
+
+Zebra's existing acceptance and integration tests will catch most hangs and deadlocks.
+
+Some tests are only run after merging to `main`. If a recently merged PR fails on `main`,
+we revert the PR, and fix the failure.
+
+Some concurrency bugs only happen intermittently. Zebra developers should run regular
+full syncs to ensure that their code doesn't cause intermittent hangs. This is
+particularly important for code that modifies Zebra's highly concurrent crates:
+* `zebrad`
+* `zebra-network`
+* `zebra-state`
+* `zebra-consensus`
+* `tower-batch`
+* `tower-fallback`
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Implementing and reviewing these constraints creates extra work for developers.
+But concurrency bugs slow down every developer, and impact users. And diagnosing
+those bugs can take a lot of developer effort.
+
+# Rationale and alternatives
```

Comment:
At this point i think we can just remove the sections that have no content. Another option is to write content for them but i am ok if we just delete them.